plugins/inputs/slurm/slurm.go

```diff
@@ -0,0 +1,666 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package slurm
+
+import (
+	"context"
+	_ "embed"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+
+	goslurm "github.com/pcolladosoto/goslurm/v0038"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type Slurm struct {
+	URL              string          `toml:"url"`
+	Username         string          `toml:"username"`
+	Token            string          `toml:"token"`
+	IgnoredEndpoints []string        `toml:"ignored_endpoints"`
+	ResponseTimeout  config.Duration `toml:"response_timeout"`
+	tls.ClientConfig
+
+	client      *goslurm.APIClient
+	baseURL     *url.URL
+	endpointMap map[string]bool
+}
+
+func (s *Slurm) createHTTPClient(host string) *goslurm.APIClient {
+	configuration := goslurm.NewConfiguration()
+	configuration.Host = host
+	configuration.Scheme = "http"
+	configuration.UserAgent = "Telegraf Metrics Agent"
+	configuration.HTTPClient = &http.Client{
+		Timeout: time.Duration(s.ResponseTimeout),
+	}
+
+	return goslurm.NewAPIClient(configuration)
+}
+
+func (s *Slurm) createHTTPSClient(host string) (*goslurm.APIClient, error) {
+	tlsCfg, err := s.ClientConfig.TLSConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	configuration := goslurm.NewConfiguration()
+	configuration.Host = host
+	configuration.Scheme = "https"
+	configuration.UserAgent = "Telegraf Metrics Agent"
+	configuration.HTTPClient = &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: tlsCfg,
+		},
+		Timeout: time.Duration(s.ResponseTimeout),
+	}
+
+	return goslurm.NewAPIClient(configuration), nil
+}
+
+func (*Slurm) SampleConfig() string {
+	return sampleConfig
+}
+
+func (s *Slurm) Init() error {
+	if s.ResponseTimeout < config.Duration(time.Second) {
+		s.ResponseTimeout = config.Duration(time.Second * 5)
+	}
```

Comment:
So you can set it to 2 or 3 seconds, but anything lower than a second will be set to 5s?

Reply:
The idea here was to configure a default timeout of 5 seconds. If no timeout is specified on the configuration it'll default to 0 seconds and the condition will be true.

I can however set the default timeout to 1 second if you prefer.

Reply:
You should keep 5 seconds as the default, but what you can do is check if the timeout is equal to 0. That means the user did not set it and in that case set it to 5 seconds. This way a user could change the value to less than 5 seconds, even if that is probably not a good idea.

Reply:
Yeah... you're completely right! I tremendously appreciate these 'user-perspective' comments as I'm not really used to taking that point of view...

As you proposed, I'm explicitly checking for `0` now to configure a default `5s` timeout. Otherwise, the user's perspective will be honoured.

This has been added on 54a9fb235c7bd933ba872d2b6c5467c13437d650.